packages/aws-cdk/lib/api/cxapp/exec.ts

```diff
@@ -214,14 +215,34 @@ export async function prepareContext(config: Configuration, env: { [key: string]
     context[cxapi.DISABLE_ASSET_STAGING_CONTEXT] = true;
   }
 
-  const bundlingStacks = config.settings.get(['bundlingStacks']) ?? ['**'];
+  const bundlingStacks = setBundlingStacks(config);
   context[cxapi.BUNDLING_STACKS] = bundlingStacks;
 
   debug('context:', context);
 
   return context;
 }
 
+function setBundlingStacks(config: Configuration) {
+  let bundlingStacks: string[];
+  if (config.command && [
+    Command.DEPLOY,
+    Command.DIFF,
+    Command.SYNTH,
+    Command.SYNTHESIZE,
+    Command.WATCH,
+  ].includes(config.command)) {
+  // If we deploy, diff, synth or watch a list of stacks exclusively we skip
+  // bundling for all other stacks.
+    bundlingStacks = config.settings.get(['exclusively'])
+      ? config.settings.get(['STACKS']) ?? ['**']
+      : ['**'];
+  } else { // Skip bundling for all stacks
+    bundlingStacks = [];
+  }
```

Comment:
The else is unnecessary if you assign `bundlingStacks` to the empty array earlier: `let bundlingStacks = new Array<string>();`